<launch>
<node pkg="tf" type="static_transform_publisher" name="base_link_broadcaster" args="0 0 0 0 0 0 base_footprint base_link 100"/>


  
  <node pkg="ntnu_fieldflux" type="CANMsgHandler" name="CAN" output="screen">
	<param name="enc_left_sub" value="/fmInformation/encoder_left" />
	<param name="enc_right_sub" value="/fmInformation/encoder_right" />
	
	<param name="controller_subscriber" type="string" value="/fmController/output" />


 </node>
  <node pkg="joy" type="joy_node" name="joy_input">
	<param name="dev" type="string" value="/dev/input/js0"/>
	<param name="deadzone" value="0.10"/>
  </node>

	    	   			
  <node pkg="differential_odometry" type="differential_odometry_node" name="differential_odometry" output="screen">
	<param name="enc_left_sub" value="/fmInformation/encoder_left" />
	<param name="enc_right_sub" value="/fmInformation/encoder_right" />
	<param name="imu_sub" value="/imu/data" />
	<param name="odom_pub" value="/fmKnowledge/odom" />
	<param name="yaw_angle_source" value="XXimu_angular_velocity"></param>  
	<param name="imu_angular_velocity_yaw_axis" value="-z"></param> 
	<param name="vehicle_frame" value="base_footprint"></param>      
	<param name="odom_estimate_frame" value="/odom_combined"></param>
	<param name="diff_steer_wheel_radius" value="0.29124"/>
	<param name="diff_steer_wheel_ticks_per_rev" value="83333"/>
	<param name="diff_steer_wheel_distance" value="1.88"/>      
  </node>


	<node pkg="ntnu_fieldflux" name="potentialFieldController" type="potentialFieldController" output="screen">
		<param name="subscribe_topic" type="string" value="/fmKnowledge/odom"/>
		<param name="v_max" type="double" value="2.0"/>
		<param name="goal_x" type="double" value="-10"/>
		<param name="goal_y" type="double" value="160"/>
		<param name="goal_yaw" type="double" value="0"/>
	</node>

</launch>

